membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -113,11 +114,10 @@ class AccountController extends LazyLogging {
       freeOrPaidSub <- OptionEither(tp.subService.either[F, P](contact).map(_.leftMap(message => s"couldn't read sub from zuora for crmId ${contact.salesforceAccountId} due to $message")))
       details <- OptionEither.liftOption(tp.paymentService.paymentDetails(freeOrPaidSub).map(\/.right))
       sub = freeOrPaidSub.fold(identity, identity)
```

Comment:
Can we retain the 'recover' here even though we've swapped the call?

Reply:
Good spot